src/services/balancer/pools/weighted-pool.service.ts

```diff
@@ -48,29 +62,43 @@ export default class WeightedPoolService {
   ): Promise<TransactionResponse> {
     if (!owner.length) return Promise.reject('No pool owner specified');
 
-    const weightedPoolFactoryAddress =
-      configService.network.addresses.weightedPoolFactory;
+    const weightedPoolFactoryAbi =
+      this.version === FactoryVersion.V2
+        ? WeightedPoolFactoryV2Abi
+        : WeightedPoolFactory__factory.abi;
 
     const tokenAddresses: Address[] = tokens.map((token: PoolSeedToken) => {
       return token.tokenAddress;
     });
 
     const seedTokens = this.calculateTokenWeights(tokens);
     const swapFeeScaled = scale(new BigNumber(swapFee), 18);
-
-    const params = [
-      name,
-      symbol,
-      tokenAddresses,
-      seedTokens,
-      swapFeeScaled.toString(),
-      owner,
-    ];
+    const rateProviders = Array(tokenAddresses.length).fill(POOLS.ZeroAddress);
+
+    const params =
+      this.version === FactoryVersion.V2
+        ? [
+            name,
+            symbol,
+            tokenAddresses,
+            seedTokens,
+            rateProviders,
+            swapFeeScaled.toString(),
+            owner,
+          ]
+        : [
+            name,
+            symbol,
+            tokenAddresses,
+            seedTokens,
+            swapFeeScaled.toString(),
+            owner,
+          ];
```

Comment:
Does V3 not expect the same params as V2? Is it not only V1 that doesn't accept the rateProviders?

Reply:
Updated